sdk-ingestion/src/main/java/care/data4life/sdk/Data4LifeClient.java

```diff
@@ -43,27 +45,31 @@
     private static final String DUMMY_CLIENT_SECRET = "secret";
     private static final String DUMMY_REDIRECT_URL = "dummy";
     /* Since this client get an OAuth access token directly and never makes
-    * any requests to the OAuth endpoint, we just add a single, dummy scope. */
+     * any requests to the OAuth endpoint, we just add a single, dummy scope. */
     private static final Set<String> DUMMY_SCOPES = new HashSet<String>(Arrays.asList("fakescope"));
 
+
     /**
      * Hidden constructor for the ingestion client.
      *
      * @param alias         Alias
      * @param userService   User service
      * @param recordService Record service
-     * @param errorHandler  Error handler
+     * @param callHandler   Call handler
      */
     protected Data4LifeClient(String alias,
                               UserService userService,
                               RecordService recordService,
-                              SdkContract.ErrorHandler errorHandler) {
-        super(alias, userService, recordService, errorHandler);
+                              CallHandler callHandler) {
+        super(alias, userService, recordService, callHandler,
+                Data4LifeClient.Companion.createLegacyDataClient(alias, userService, recordService, callHandler),
+                Data4LifeClient.Companion.createLegacyAuthClient(alias, userService, recordService, callHandler)
+        );
```

Comment:
Nitpick: This could be chopped...but it is also okish for new.

Reply:
it could